tests/testthat/test-sector_profile.R

```diff
@@ -127,3 +127,30 @@ test_that("at company level, two matched and one unmatched products yield `value
   other <- pull(filter(out, !is.na(risk_category)), value)
   expect_equal(sort(other), c(0, 0, 2 / 3))
 })
+
+test_that("at product level, when `companies$tilt_sector` doesn't match `scenarios$sector` then `product$grouped_by` is `NA`", {
+  # This is what I think an analyst would say:
+  # > A product's tilt_sector with unmatched scenarios's sector yields a "missing benchmark"
+  #
+  # Translation to tiltIndicator's code:
+  # * given: the value "b" of `companies$clustered`
+  # * when: the corresponding value of `companies$tilt_sector` doesn't match any value of `scenarios$sector`
+  # * then: the result at product level has the value `NA` in `product$grouped_by`
+
+  # Based on this GoogleSheet row:
+  # https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064&range=A5:I5
+  # styler: off
+  companies <- tribble(
+    ~companies_id, ~clustered, ~activity_uuid_product_uuid, ~tilt_sector, ~tilt_subsector,       ~type,     ~sector,  ~subsector,
+              "a",        "b",                 "unmatched",  "unmatched",     "unmatched", "unmatched", "unmatched", "unmatched",
+  )
+  scenarios <- tribble(
+    ~sector,   ~subsector,  ~year, ~reductions, ~type, ~scenario,
+    "total",     "energy",   2050,         1.0, "ipr",       "a",
+  )
+  # styler: on
+
+  product <- sector_profile(companies, scenarios) |> unnest_product()
+
+  expect_true(is.na(product$grouped_by))
+})
```

Comment:
@Tilmon, 

Below I unpack a piece of your https://github.com/2DegreesInvesting/tiltIndicator/pull/739#issuecomment-1977426095. 

I turned my understanding into a test (highlighted code), and it's corresponding reprex for your convenience. Please review and correct any misunderstanding.

<details/>
<summary/>
reprex
</summary>

``` r
devtools::load_all()
#> ℹ Loading tiltIndicator

companies <- tribble(
  ~companies_id, ~clustered, ~activity_uuid_product_uuid, ~tilt_sector, ~tilt_subsector,       ~type,     ~sector,  ~subsector,
            "a",        "b",                 "unmatched",  "unmatched",     "unmatched", "unmatched", "unmatched", "unmatched",
)
scenarios <- tribble(
  ~sector,   ~subsector,  ~year, ~reductions, ~type, ~scenario,
  "total",     "energy",   2050,         1.0, "ipr",       "a",
)
# styler: on

product <- sector_profile(companies, scenarios) |> unnest_product()
product
#> # A tibble: 1 × 11
#>   companies_id grouped_by risk_category profile_ranking clustered
#>   <chr>        <chr>      <chr>                   <dbl> <chr>    
#> 1 a            <NA>       <NA>                       NA b        
#> # ℹ 6 more variables: activity_uuid_product_uuid <chr>, tilt_sector <chr>,
#> #   scenario <chr>, year <dbl>, type <chr>, tilt_subsector <chr>

is.na(product$grouped_by)
#> [1] TRUE
```

</details>

----

> For the `sector*()` ... what matters is:

> Whether the clustered has a tilt_sector.

When constructing toy data focus on `companies$tilt_sector`, then adjust `companies$sector` accordingly (see next).

> [If it doesn't have a tilt_sector]^1, the clustered [will neither have a sector or subsector for any of the types]^2.

* ^1: Means {if the value of `companies$tilt_sector` doesn't match any
`scenarios$sector`} ([example](https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064&range=A5:I5)).

* ^2: Means {yields `product$grouped_by == NA`} ([example](https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064&range=B19)).

Clarifications:

> If the tilt_sector ... is [missing]^3.

* ^3: Here, "missing" does not mean `NA` neither it means the absence of a row. What this means is ^1 above.

> no scenario sector = [no result at all]^4.

* ^4: Here, "no result at all" doesn't mean no output. It means ^2.

> clustered b [has no tilt_sector]^5 ... and will hence lead to [no results]^6

* ^5: Here, "has no tilt_sector" doesn't mean `NA` or absence. It means ^1.

* ^6: Here, "no results" doesn't mean no output. It means ^2.